### DISCUSSION TITLE
Can we use Artillery-Playwright to Test Single Page Applications?

### DISCUSSION THREAD

**Discussion by PranavKutralingam** [artilleryio/artillery#1924] [Artillery]
Hello,

I am attempting to use artillery-playwright to test the UI performance of a single-page application (the URL does not change even with changing pages and objects). 
Currently, I am only getting the average performance of the whole flow and not for individual pages.

Is there any provision within Artillery-Playwright to capture the performance of new pages or objects during the execution of the flow?

**Comment by hassy** [comment]
hi @PranavKutralingam 👋 

What sort of performance metrics would you be looking at tracking for a SPA? Out of the Web Vitals metrics we report, which ones would still make sense in the context of your SPA?
